xarray/core/indexing.py

```diff
@@ -135,11 +135,27 @@ def _asarray_tuplesafe(values):
     return result
 
 
+def _is_nested_tuple(tup, index):
+    """Check for a compatible nested tuple and multiindex (taken from
+    pandas.core.indexing.is_nested_tuple).
```

Comment:
This is such a weird function the way it's currently written. Why not make this:
```python
def _is_nested_tuple(possible_tuple):
    return (isinstance(possible_tuple, tuple)
            and any(isinstance(value, (tuple, list, slice)
                    for value in possible_tuple))
```

The `isinstance(index, pd.MultiIndex)` should be outside the function, since it's totally unrelated.